includes/sanitizers/class-amp-style-sanitizer.php

```diff
@@ -2130,15 +2130,32 @@ private function process_css_list( CSSList $css_list, $options ) {
 		$viewport_rules     = [];
 		$imported_font_urls = [];
 
-		foreach ( $css_list->getContents() as $css_item ) {
+		foreach ( $css_list->getContents() as $i => $css_item ) {
 			$sanitized = false;
 			if ( $css_item instanceof DeclarationBlock && empty( $options['validate_keyframes'] ) ) {
 				$validation_results = array_merge(
 					$validation_results,
 					$this->process_css_declaration_block( $css_item, $css_list, $options )
 				);
 			} elseif ( $css_item instanceof AtRuleBlockList ) {
-				if ( ! in_array( $css_item->atRuleName(), $options['allowed_at_rules'], true ) ) {
+				if (
+					'-moz-document' === $css_item->atRuleName()
+					&&
+					'url-prefix()' === $css_item->atRuleArgs()
+					&&
+					in_array( 'supports', $options['allowed_at_rules'], true )
+				) {
+					// Replace `@-moz-document url-prefix()` with `@supports (-moz-appearance:meterbar)` as an alternative
+					// way to provide Firefox-specific style rules. This is a workaround since @-moz-document is not
+					// yet allowed in AMP, and this use of @supports is another recognized Firefox-specific CSS hack,
+					// per <http://browserhacks.com/#hack-8e9b5504d9fda44ec75169381b3c3157>.
+					// For adding @-moz-document to AMP, see <https://github.com/ampproject/amphtml/issues/26406>.
+					$new_css_item = new AtRuleBlockList( 'supports', '(-moz-appearance:meterbar)' );
+					$new_css_item->setContents( $css_item->getContents() );
+					$css_list->splice( $i, 1, [ $new_css_item ] );
```

Comment:
Looks like the validation error is now being leaked. `$new_css_item` is not replacing the rule, but instead being inserted after it. We then get this:

![image](https://user-images.githubusercontent.com/16200219/96819866-3372e000-1414-11eb-8481-6538f01d5e64.png)

![image](https://user-images.githubusercontent.com/16200219/96819884-3c63b180-1414-11eb-8050-9a233c7eee11.png)


![image](https://user-images.githubusercontent.com/16200219/96819820-11795d80-1414-11eb-8284-10dcfb814a3b.png)

Reply:
That's strange. The `allowed_at_rules_retained` test shows that the `@-moz-document` is getting replaced rather than being retained.

Reply:
But I am indeed seeing the same thing as you on the frontend.

Reply:
I've managed to fix that by doing `remove` after the `replace` in a4bf60a0bd5e668c323250d0eb633d7465528f4d. But this should not have been required since the `array_splice()` should have already removed the item!?